### PR TITLE
Reorganized test files to match src folder structure

### LINKED ISSUE #160: Organize test files according to `src` folder structure
This is our current `src` structure.
Screenshot 2024-11-06 at 9 35 58 PM

This is our current `tests` structure
Screenshot 2024-11-06 at 9 36 11 PM

### Problem

Test files are not organized into folders.

### Proposed solution

Under `tests`, create a folder for each corresponding folder under `src`.

### PR DESCRIPTION
**Pull request by alisnwu**
closes #160

Screenshot 2024-11-12 at 11 23 46 AM

@sbillinge pytest passed, ready for review

### CODE REVIEW

File: tests/diffpy/utils/parsers/test_serialization.py [hunk 1]
```diff
@@ -7,20 +7,19 @@
...
-
 
 def test_load_multiple(tmp_path, datafile):
     # Load test data
     targetjson = datafile("targetjson.json")
     generatedjson = tmp_path / "generated_serialization.json"
 
-    tlm_list = os.listdir(os.path.join(tests_dir, "testdata", "dbload"))
-    tlm_list.sort()
+    dbload_dir = os.path.dirname(datafile("dbload/e1.gr"))
```
Comment (sbillinge) [inline tests/diffpy/utils/parsers/test_serialization.py:16]: then do these two lines using pathlib and pathlib globing?

File: tests/diffpy/utils/parsers/test_serialization.py [hunk 2]
```diff
@@ -7,20 +7,19 @@
...
+    tlm_list = sorted(os.listdir(dbload_dir))
+
     generated_data = None
     for hfname in tlm_list:
         # gather data using loadData
-        headerfile = os.path.normpath(os.path.join(tests_dir, "testdata", "dbload", hfname))
+        headerfile = os.path.join(dbload_dir, hfname)
```
Comment (sbillinge) [inline tests/diffpy/utils/parsers/test_serialization.py:22]: and here.

**Review by sbillinge — COMMENTED** [review]
few inlines

File: tests/conftest.py [hunk 3]
```diff
@@ -22,8 +22,9 @@ def user_filesystem(tmp_path):
 @pytest.fixture
 def datafile():
     """Fixture to dynamically load any test file."""
+    base_path = Path(__file__).parent / "testdata"  # Adjusted base path
 
     def _load(filename):
-        return "tests/testdata/" + filename
+        return str(base_path / filename)
```
[the remark this answers is not in the document]
Comment (alisnwu) [reply tests/conftest.py:28]: I was trying to match the original return of strings but the Path object also works, will edit to just return the Path object

File: tests/diffpy/utils/parsers/test_serialization.py (hunk 1 above)
Comment (alisnwu) [reply tests/diffpy/utils/parsers/test_serialization.py:16]: globing returns a tuple[ndarray, dict] to hdata but serialize_data wants a dict

File: tests/conftest.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sbillinge) [reply tests/conftest.py:28]: This work is exactly what `Path` is designed for.

File: tests/diffpy/utils/parsers/test_serialization.py (hunk 1 above)
Comment (alisnwu) [reply tests/diffpy/utils/parsers/test_serialization.py:16]: Yep, will use pathlib for both parts

**Comment by alisnwu** [comment]
@sbillinge Ready for review
Screenshot 2024-11-12 at 9 28 11 PM

**Comment by sbillinge** [comment]
Much nicer, thanks!

**Pull request merged**